jetpack.php

```diff
@@ -187,5 +187,8 @@ function jetpack_admin_missing_autoloader() {
 	return;
 }
 
+register_activation_hook( __FILE__, array( 'Jetpack', 'plugin_activation' ) );
+register_deactivation_hook( __FILE__, array( 'Jetpack', 'plugin_deactivation' ) );
```

Comment:
Just a note that in this current place, it's possible for it to not get hit if the site is on an unsupported version of WP or PHP, right? Is it better to move to the top? The problem with moving to the top is we don't know what code the hook will trigger, and it could be incompatible code which would error.

Reply:
Yes, you're right, it could trigger code that won't be compatible with the current software. We could try to  make sure we don't use anything incompatible in activation/deactivation hooks, but it would not be trivial, or easily maintainable.
Do we even need hooks in case Jetpack is not supported on the current software? The only case would be to clean up the options in case Jetpack just became unusable after an update, and the user wants to disable it.